twindb_backup/destination/s3.py

```diff
@@ -333,9 +346,21 @@ def _upload_object(self, file_obj, object_key):
         s3_transfer_config = self.get_transfer_config()
 
         LOG.debug("Starting to stream to %s", remote_name)
-        self.s3_client.upload_fileobj(file_obj, self.bucket, object_key,
-                                      Config=s3_transfer_config)
-        LOG.debug("Successfully streamed to %s", remote_name)
+        retry_timeout = time.time() + S3_READ_TIMEOUT
+        retry_interval = 2
+        while time.time() < retry_timeout:
+            try:
+                self.s3_client.upload_fileobj(file_obj,
```

Comment:
How does this work when the `upload_fileobj` function fails in the middle of the transfer. The code will retry again. However, there is no way to seek back to the start of the `file_obj` that is passed to the function as it is just a stream.